### PR TITLE
Handle bad camera_info messages

### PR DESCRIPTION
**Pull request by rohbotics**
Should prevent the situation in #90 from crashing with a segfault.

### CODE REVIEW

File: aruco_detect/src/aruco_detect.cpp [hunk 1]
```diff
@@ -238,18 +238,23 @@ void FiducialsNode::camInfoCallback(const sensor_msgs::CameraInfo::ConstPtr& msg
         return;
     }
 
-    for (int i=0; i<3; i++) {
-        for (int j=0; j<3; j++) {
-            cameraMatrix.at<double>(i, j) = msg->K[i*3+j];
+    if (msg->distortion_model != "") {
+        for (int i=0; i<3; i++) {
```
Comment (jimvaughan) [inline aruco_detect/src/aruco_detect.cpp]: This doesn't look right: there should be a msg->K independently of whether there is a distortion model

**Review by jimvaughan — CHANGES_REQUESTED** [review]

File: aruco_detect/src/aruco_detect.cpp (hunk 1 above)
Comment (jimvaughan) [reply aruco_detect/src/aruco_detect.cpp]: Thanks for doing the PR

The nodes should work with a K matrix but no distortion model, but should exit with an error if K is not provided.

Comment (rohbotics) [reply aruco_detect/src/aruco_detect.cpp]: Maybe I am misunderstanding how the calibration works, but doesn't the K matrix not make sense if  there is no distortion model? Or is that only the D matrix?

**Comment by jim-v** [comment]
`K` is the intrinsic matrix http://ksimek.github.io/2013/08/13/intrinsic/ containing  the focal lengths and image center.  It is needed to do any 2D -> 3D mapping.  

`D` contains the distortion coefficients which are used to undistort the image.  The undistortion is (or should be) optional

**Comment by rohbotics** [comment]
Thanks for the link, it was pretty useful.

I am now checking for a all 0 K matrix instead of an empty distortion model.

**Review by jim-v — APPROVED** [review]

**Pull request merged**